hudi-common/src/main/java/org/apache/hudi/common/util/hash/BucketIndexUtil.java

```diff
@@ -37,9 +38,12 @@ public class BucketIndexUtil {
    */
   public static Functions.Function3<Integer, String, Integer, Integer> getPartitionIndexFunc(int parallelism) {
     return (bucketNum, partition, curBucket) -> {
```

Comment:
Maybe we can change `Functions.Function3<Integer, String, Integer, Integer>` to `Functions.Function3<Long, String, Integer, Integer>`

Reply:
> Maybe we can change `Functions.Function3<Integer, String, Integer, Integer>` to `Functions.Function3<Long, String, Integer, Integer>`

Anyway, bucket-num is expressed using the int type in all code paths. I think this change is not necessary. After all, we just want to avoid overflow during the calculation process. The input parameters and the output partition id itself all need to be of int type.